runtime/compiler/runtime/asmprotos.h

```diff
@@ -64,7 +64,7 @@ extern "C" {
  * is set for OS variants that currently use NASM and is used to
  * decide the version of helper names to use.
  */
-#if defined(OSX) || (defined(LINUX) && defined(TR_HOST_X86) && defined(TR_HOST_64BIT))
+#if (defined(TR_HOST_X86) && defined(TR_HOST_64BIT)) && (defined(OSX) || defined(LINUX) || defined(WINDOWS))
```

Comment:
Should we remove `defined(OSX) || defined(LINUX) || defined(WINDOWS)`? OpenJ9 now only supports these three platforms, and hence this `defined` seems unnecessary.